pkg/web/apps/apps_repository.go

```diff
@@ -11,4 +11,10 @@ type Repository interface {
 	GetAppVersionsByAppID(ID string) (*[]models.Version, error)
 	UpdateAppVersions(versions []models.Version) error
 	DisableAllAppVersionsByAppID(appID string, message string) error
+	GetVersionByAppIDAndVersion(appID string, versionNumber string) (*models.Version, error)
+	GetDeviceByDeviceIDAndAppID(deviceID string, appID string) (*models.Device, error)
+	GetDeviceByVersionAndAppID(versionID string, appID string) (*models.Device, error)
+	GetAppByAppID(appID string) (*models.App, error)
+	InsertVersionOrUpdateNumOfAppLaunches(version *models.Version) error
+	InsertDeviceOrUpdateVersionID(device models.Device) error
```

Comment:
@craicoverflow it shows not be a good practice have a method/func which does more than one action. I mean you are losing cohesion when you do a func as InsertOrUpdate. I'd like to recommend you change it anyway.

Reply:
Hi @camilamacedo86 I would normally agree but this is done with a SQL feature known as `UPSERT`. Therefore it is really doing one action.

https://wiki.postgresql.org/wiki/UPSERT

Reply:
I know that it is possible to be done in the database. However, please check my suggestion https://github.com/aerogear/mobile-security-service/pull/76#discussion_r which also will solve this scenario.